PR: https://github.com/odoo/odoo/pull/200345

From: 64d5c0be671c1926dff8d1a3ca6e275add037922
From: Pieter Claeys (clpi)
Date: 2025-04-17 09:39:05

Breaking data model changes scores: del:8 + add:0, change matches:
-    worksheet_type = fields.Selection([
-    note = fields.Html('Description')
-    worksheet = fields.Binary('PDF')
-    worksheet_google_slide = fields.Char('Google Slide', help="Paste the url of your Google Slide. Make sure the access to the document is public.", tracking=True)
-    worksheet = fields.Binary(
-    worksheet_type = fields.Selection(
-    worksheet_google_slide = fields.Char(
-    operation_note = fields.Html("Description", related='operation_id.note', readonly=True)

Total Changes: 14

[IMP] mrp: simplify quality check documents

We remove all worksheet types on the operation level.
We only keep instructions (text) and single PDF upload at the quality
point level.

task-4484941

Part-of: odoo/odoo#200345
Related: odoo/enterprise#80469
Related: odoo/upgrade#7331
Signed-off-by: Arnold Moyaux (arm) <arm@odoo.com>

================================= pseudo patch: =================================

--- a/addons/mrp/models/mrp_routing.py
+++ b/addons/mrp/models/mrp_routing.py
@@ -24,13 +24,6 @@ class MrpRoutingWorkcenter(models.Model):
         'mrp.bom', 'Bill of Material',
         index=True, ondelete='cascade', required=True, check_company=True)
     company_id = fields.Many2one('res.company', 'Company', related='bom_id.company_id')
-    worksheet_type = fields.Selection([
-        ('pdf', 'PDF'), ('google_slide', 'Google Slide'), ('text', 'Text')],
-        string="Worksheet", default="text", tracking=True
-    )
-    note = fields.Html('Description')
-    worksheet = fields.Binary('PDF')
-    worksheet_google_slide = fields.Char('Google Slide', help="Paste the url of your Google Slide. Make sure the access to the document is public.", tracking=True)
     time_mode = fields.Selection([
         ('auto', 'Compute based on tracked time'),
         ('manual', 'Set duration manually')], string='Duration Computation',

--- a/addons/mrp/models/mrp_workorder.py
+++ b/addons/mrp/models/mrp_workorder.py
@@ -99,13 +99,6 @@ class MrpWorkorder(models.Model):
     operation_id = fields.Many2one(
         'mrp.routing.workcenter', 'Operation', check_company=True, index='btree_not_null')
         # Should be used differently as BoM can change in the meantime
-    worksheet = fields.Binary(
-        'Worksheet', related='operation_id.worksheet', readonly=True)
-    worksheet_type = fields.Selection(
-        string='Worksheet Type', related='operation_id.worksheet_type', readonly=True)
-    worksheet_google_slide = fields.Char(
-        'Worksheet URL', related='operation_id.worksheet_google_slide', readonly=True)
-    operation_note = fields.Html("Description", related='operation_id.note', readonly=True)
     move_raw_ids = fields.One2many(
         'stock.move', 'workorder_id', 'Raw Moves',
         domain=[('raw_material_production_id', '!=', False), ('production_id', '=', False)])
